### PR TITLE
If provisioning profile name is set, search for that one

### PR DESCRIPTION
**Pull request by champo**
The parameter was used only when creating a new profile. I've added the check when choosing a profile to renew or download directly.

### CODE REVIEW

**Comment by KrauseFx** [comment]
Not sure I completely understand what this should solve?

**Comment by champo** [comment]
Right now for me, sigh is always downloading the `iOS Team Provisioning Profile` for most apps. However, since I have devices belonging to several clients in my account each app has it's own manually created profile with my company's devices and the clients. This change allows me to hand pick the correct profile with sigh.

Also, it struck me as weird when I tried using that parameter and it didn't download the profile with that name.

**Comment by champo** [comment]
Hey, I was wondering what your opinion on this is? This is kinda blocking me from having automatic builds, so if you don't like it I can add it as a custom action on my end.

**Comment by KrauseFx** [comment]
Sorry, I missed the notifications, I'll check it right now.

File: lib/sigh/developer_center.rb [hunk 1]
```diff
@@ -58,6 +60,9 @@ def maintain_app_certificate(force = nil)
           details = profile_details(current_cert['provisioningProfileId'])
 
           if details['provisioningProfile']['appId']['identifier'] == Sigh.config[:app_identifier]
+
+            next if profile_name != nil && details['provisioningProfile']['name'] != profile_name
```
Comment (KrauseFx) [inline lib/sigh/developer_center.rb]: Please change to 
```ruby
next if profile_name && details['provisioningProfile']['name'] != profile_name
```

**Comment by KrauseFx** [comment]
Once you changed this, I'll merge the pull request :

**Comment by champo** [comment]
Just realized the property name changed on master a few hours ago, I'll update in a sec.

**Comment by KrauseFx** [comment]
Hu, yah, sorry about that! Thanks for checking and correcting it :+1:

**Comment by KrauseFx** [comment]
Thanks for your contribution @champo :+1:

**Pull request merged**